source/common/router/router.h

```diff
@@ -276,6 +312,11 @@ class Filter : Logger::Loggable<Logger::Id::router>,
   uint32_t buffer_limit_{0};
   bool stream_destroyed_{};
 
+  // list of cookies to add to upstream headers
+  // 'mutable' is safe because this is only read in onUpstreamHeaders(), which
+  // is not const
+  mutable std::vector<std::string> downstream_set_cookies_;
```

Comment:
Do you know if the default allocator/constructor for vector pre-allocates space? I would be unfortunate to allocate anything in the common case. Wondering if this should be a pointer to a vector and you can allocate it when needed?

Reply:
The clang and GCC standard library vectors don't pre-allocate.